src/common/snippets/src/pass/mha_tokenization.cpp

```diff
@@ -0,0 +1,390 @@
+                    // If Constant has one consumer - target node, we add Constant to body_inputs
+                    // If Constant has several consumers, we should check that all these consumers are inside Subgraph body
+                    // and if all of them are inside body, we can explicitly add Constant to the body_inputs, otherwise we should
+                    // make a copy and add copy of Constant to body_inputs
+                    // For example, this case is especially valid for Transposes nodes
+                    //              (several Transposes have the same order so there can be the common Constant with this order)
+                    if (constant->get_output_target_inputs(0).size() == 1) {
+                        body_inputs.push_back(input.get_source_output());
+                    } else {
+                        const auto constant_consumers = constant->get_output_target_inputs(0);
+                        bool all_consumers_are_inside = std::all_of(constant_consumers.begin(), constant_consumers.end(),
+                                                                    [&ordered_ops](const ngraph::Input<ngraph::Node>& input) {
+                                                                        return std::find(ordered_ops.begin(), ordered_ops.end(),
+                                                                                         input.get_node()->shared_from_this()) != ordered_ops.end();
+                                                                    });
+                        if (all_consumers_are_inside) {
+                            body_inputs.push_back(input.get_source_output());
+                        } else {
+                            const auto constant_copy = constant->clone_with_new_inputs({});
+                            node->set_argument(input.get_index(), constant_copy);
+                            body_inputs.push_back(constant_copy);
+                        }
+                    }
+                } else if (std::find(ordered_ops.begin(), ordered_ops.end(), parent) == ordered_ops.end()) {
+                    auto parameter = std::make_shared<ngraph::opset1::Parameter>(input.get_element_type(), input.get_partial_shape());
+                    body_parameters.push_back(parameter);
+                    body_parameters.back()->set_friendly_name(input.get_node()->get_friendly_name());
+                    body_inputs.push_back(parameter->output(0));
+
+                    subgraph_inputs.push_back(input.get_source_output());
+
+                    node->input(i).replace_source_output(parameter);
+                }
+            }
+        };
+
+        for (const auto& op : ordered_ops) {
+            create_body_inputs(op);
+            op->clear_control_dependencies();
+            fused_names += op->get_friendly_name() + ",";
+        }
+
+        const auto last_node = ordered_ops.back();
+        for (const auto output : last_node->outputs()) {
```

Comment:
Minor: please use const reference here and on L350

Reply:
Sure, thanks